controllers/rcj_soccer_referee_supervisor/referee/supervisor.py

```diff
@@ -316,18 +322,59 @@ def reset_positions(self):
         Reset the positions of the ball as well as the robots to the initial
         position.
         """
-
         self.reset_ball_position()
 
         # reset the robot positions
         for robot in ROBOT_NAMES:
             self.reset_robot_position(robot)
 
-    def reset_ball_position(self):
+    def set_robot_position(self, robot_name: str, position: List[float]):
+        """Set the position of a robot.
+
+        Args:
+            robot_name (str): The robot we are moving
+            position (list of floats): The actual position
+        """
+        tr_field = self.robot_translation_fields[robot_name]
+        tr_field.setSFVec3f(position)
+        self.robot_translation[robot_name] = position
+
+    def set_robot_rotation(self, robot_name: str, rotation: List[float]):
+        """Set the rotation of a robot.
+
+        Args:
+            robot_name (str): The robot we are rotating
+            rotation (list of floats): The actual rotation
+        """
+        rot_field = self.robot_rotation_fields[robot_name]
+        rot_field.setSFRotation(rotation)
+
+    def reset_robot_velocity(self, robot_name: str):
+        """Reset the robot's velocity.
+
+        Args:
+            robot_name (str): The robot we set the velocity for
+        """
+        self.getFromDef(robot_name).setVelocity([0, 0, 0, 0, 0, 0])
+
+    def set_ball_position(self, position: List[float]):
+        """Set the position of the ball.
+
+        Args:
+            position (list of floats): The actual position
+        """
         ball_translation_field = self.ball.getField("translation")
-        ball_translation_field.setSFVec3f(BALL_INITIAL_TRANSLATION)
+        ball_translation_field.setSFVec3f(position)
+        self.reset_ball_velocity()
+        self.ball_translation = position
 
+    def reset_ball_velocity(self):
+        """Reset the ball's velocity."""
         self.ball.setVelocity([0, 0, 0, 0, 0, 0])
+
+    def reset_ball_position(self):
```

Comment:
@Adman I think we should also call `resetPhysics` in here (as well as in the `reset_robot_position`) -- here is the webots issue which discusses what our issue #55  also noticed

https://github.com/cyberbotics/webots/issues/1118

Reply:
Thanks, should be fixed. @mrshu could you please recheck again?